### PR TITLE
[www/nginx] #1561 advanced buffering

### PR DESCRIPTION
**Pull request by fabianfrz**
also clean up some whitespaces.

### CODE REVIEW

File: www/nginx/src/opnsense/service/templates/OPNsense/Nginx/location.conf [hunk 1]
```diff
@@ -190,6 +199,6 @@ location {{ location.matchtype }} {{ location.urlpattern }} {
     proxy_hide_header {{ our_header }};
 {%   endfor %}
 {% endif %}
-{% endif %}
+{% endif %}{# honeypod #}
```
Comment (fichtner) [inline www/nginx/src/opnsense/service/templates/OPNsense/Nginx/location.conf]: not a typo?

File: www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+use \Phalcon\Validation\Message;
+
+/**
+ * a very specific nginx check - not reusable
+ *
+ * Class NgxBusyBufferConstraint
+ * @package OPNsense\Nginx\Constraints
+ */
+class NgxBusyBufferConstraint extends BaseConstraint
```
Comment (fichtner) [inline www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php]: Clever solution 👍

Comment (fabianfrz) [reply www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php]: Thanks - seems to be the only usable callback for model validation.

The problem here is that the minimum and maximum values have to be calculated and depend on other values. What I hated most about that is that it was not in the online docs but raised an emerg error in the nginx config.

Comment (AdSchellevis) [reply www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php]: @fabianfrz you can also bootstrap custom types per app model for more complex validations, like https://github.com/opnsense/core/blob/98673a0b30b48d68cee9bb0f52e0c68d1d6de51d/src/opnsense/mvc/app/models/OPNsense/Firewall/Alias.xml#L49

Comment (fabianfrz) [reply www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php]: @AdSchellevis I looked for that and did not find that. I think this should also go to the docs.

Comment (AdSchellevis) [reply www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php]: that would be practical indeed

**Pull request merged**